IMC/chatbot/Chatbot.md

```diff
@@ -0,0 +1,167 @@
+
+First, let’s look into why we might need a Chatbot:
+
+- **Navigation:** You want to help a user navigate through a website.
+- **Information:** Search for news relevant to you.
+- **Customer Assistance:** Your business requires a lot of customer assistance and you want to automate responses to complaints that do not require urgent attention.
+- **Social-Automation:** Automate restaurant reservations.
+- **Curiosity:** You just want to explore the limits of computer intelligence (or trickery, see the image below ⬇️)
+
+![img](images/2.gif)
+
+### Why are Chatbots important for business?
+
+There are a massive amount of active users online these days. Chatbots make it much easier to offer customer service. They can interact with multiple users at the same time and work 24 hours a day. They provide users with helpful information to enhance customer experience while reducing the cost of traditional customer service, and prevent your team from always answering the same FAQs.
+
+As you can see, the advantages of a Chatbot are far-reaching. Fortunately for us, there are frameworks and services which make it relatively easy to create information-based bots.
+
+
+### What is Google Dialogflow?
+
+For this tutorial, we will be using Google's **DialogFlow**. DialogFlow is a powerful NLU API that was acquired and is now developed by Google. It gives users new ways to interact with your product by building engaging voice and text-based conversational interfaces, such as voice apps and Chatbots, powered by AI. You can connect with users on your website, mobile app, Google Assistant, Amazon Alexa, Facebook Messenger, and other popular platforms and devices.
+
+What makes DialogFlow so useful and popular:
+
+- Powered by Google’s Machine Learning
+- Built on Google infrastructure
+- Optimized for the Google Assistant
+- It is very user friendly, intuitive and is one of the best NLP models out there
+
+## Building Blocks
+
+**Agents:**
+DialogFlow allows you to make Natural language understanding (NLU) modules called agents. The agent is the main component of your bot. It connects to your backend and provides it with your custom logic. Agents are made up of intents. 
+
+**Intents:**
+Intents are actions that a user can perform on your agent. You can set up as many intents as you want for your agent. When a user says something, DialogFlow will match the expression with one of the intents that you created. For example, you may have a “Shipping” intent. When a user asks “What are the shipping methods?“ the Chatbot will match it with the “Shipping” intent. The Chatbot knows which intent to call by using entities.
```

Comment:
would it be helpful to include something about how the intent creates the response to the user? maybe after
 "When a user asks “What are the shipping methods?“ the Chatbot will match it with the “Shipping” intent", you can say "so the correct response can be delivered to the Chatbot user." 

I'm not sure if that's how it works, but it would be helpful to include somewhere in the building blocks how a response is made

Reply:
I added some stuff to help understand the concept.